sql/src/test/java/io/crate/testing/LuceneDocCollectorProvider.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    private String extractIndexName(Analysis analysis) {
+        return analysis.analyzedStatement().accept(new AnalyzedStatementVisitor<Object, String>() {
+            @Override
+            protected String visitSelectStatement(SelectAnalyzedStatement stmt, Object context) {
+                if (stmt.relation() instanceof QueriedTableRelation) {
+                    QueriedTableRelation<?> tableRelation = (QueriedTableRelation<?>) stmt.relation();
+                    return tableRelation.tableRelation().tableInfo().ident().esName();
+                }
+                throw new IllegalArgumentException("cannot get LuceneDocCollectors for Analysis");
+            }
+        } , null);
+    }
+
+    private Iterable<CrateCollector> createNodeCollectors(String nodeName, String indexName, Analysis analysis, RowReceiver downstream) {
+        IndexService indexService;
+        IndicesService indicesService;
+            try {
+                indicesService = cluster.getInstance(IndicesService.class, nodeName);
+                indexService = indicesService.indexServiceSafe(indexName);
+            } catch (IndexMissingException e) {
+                return Collections.emptyList();
+            }
+
+        JobContextService jobContextService = indexService.shardInjectorSafe(0).getInstance(JobContextService.class);
+        ClusterService clusterService = cluster.getInstance(ClusterService.class, nodeName);
+        MapSideDataCollectOperation collectOperation = cluster.getInstance(MapSideDataCollectOperation.class, nodeName);
+
+        PlannedAnalyzedRelation plannedAnalyzedRelation = queryAndFetchConsumer.consume(
+                analysis.rootRelation(),
+                new ConsumerContext(analysis.rootRelation(), new Planner.Context(clusterService, UUID.randomUUID(), null)));
+        CollectPhase collectPhase = ((CollectAndMerge) plannedAnalyzedRelation.plan()).collectPhase();
+
+        SharedShardContexts sharedShardContexts = new SharedShardContexts(indicesService);
+        JobExecutionContext.Builder builder = jobContextService.newBuilder(collectPhase.jobId());
+        JobCollectContext jobCollectContext = new JobCollectContext(collectPhase, collectOperation, RAM_ACCOUNTING_CONTEXT, downstream, sharedShardContexts);
+        collectContexts.add(jobCollectContext);
+        builder.addSubContext(jobCollectContext);
+        jobContextService.createContext(builder);
+
+        return collectOperation.createCollectors(
+            collectPhase,
+            downstream,
+            jobCollectContext
+        );
+    }
+
+    public Iterable<CrateCollector> createCollectors(String statement, RowReceiver downstream, Object ... args) {
+        Analysis analysis = analyzer.analyze(
+                SqlParser.createStatement(statement), new ParameterContext(args, new Object[0][], null));
+        String indexName = extractIndexName(analysis);
```

Comment:
you could create the CollectPhase first which contains the routing with the nodeNames